src/main.rs

```diff
@@ -1,58 +1,45 @@
-// src/main.rs
+mod node;
+mod routing_table;
+mod kbucket;
+mod message;
+mod utils;
+use log::{info, warn, debug, error};
 
-use kademlia_mvp::*;
-use tokio::time::Duration;
-use tokio::time::sleep;
-use log::{debug, error, info, warn};
+use node::KademliaNode;
+// use tokio::net::SocketAddr;
+use std::net::SocketAddr;
 
 #[tokio::main]
-async fn main() -> Result<(), Box<dyn std::error::Error>> {
-    // Initialize logging
-    env_logger::init();
+async fn main() -> std::io::Result<()> {
+    env_logger::init(); // Initialize the logger
+    info!("Starting Kademlia node...");
 
-    // Create two nodes
-    let addr1 = "127.0.0.1:8001".parse()?;
-    let addr2 = "127.0.0.1:8002".parse()?;
-    let mut node1 = KademliaNode::new(addr1).await;
-    let node2 = KademliaNode::new(addr2).await;
+    let addr: SocketAddr = "127.0.0.1:33334".parse().expect("Failed to parse socket address");
+    info!("Parsed socket address: {}", addr);
 
-    // Log the creation of nodes
-    println!("Created node1 with address: {}", addr1);
-    println!("Created node2 with address: {}", addr2);
-
-    // Start node2 in the background
-    tokio::spawn(async move {
-        if let Err(e) = node2.start().await {
-            println!("Failed to start node2: {:?}", e);
-        } else {
-            println!("Node2 started successfully");
+    let (mut node, shutdown_sender) = match KademliaNode::new(addr).await {
+        Ok((node, sender)) => {
+            info!("Successfully created Kademlia node.");
+            (node, sender)
         }
-    });
-
-    // Allow some time for node2 to start
-    sleep(Duration::from_millis(100)).await;
-    println!("Slept for 100ms to allow node2 to start");
+        Err(e) => {
+            error!("Failed to create Kademlia node: {:?}", e);
+            return Err(e);
+        }
+    };
 
-    // Node1 pings Node2
-    println!("Node1 attempting to ping Node2 at address: {}", addr2);
-    if let Err(e) = node1.ping(addr2).await {
-        println!("Node1 failed to ping Node2: {:?}", e);
-    } else {
-        println!("Node1 successfully pinged Node2");
+    info!("Bootstrapping node...");
+    if let Err(e) = node.bootstrap().await {
+        error!("Bootstrap failed: {:?}", e);
```

Comment:
unless you want to log here, `.map_err` might be cleaner. you could also log inside the `.map_err`